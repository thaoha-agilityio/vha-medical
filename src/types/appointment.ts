import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export type AppointmentStatus = 0 | 1 | 2 | 3;

export interface AppointmentStatusOption {
  key: string;
  label: string;
  value: AppointmentStatus;
}

export interface AppointmentModel {
  startTime: string;
  durationTime: string;
  status: AppointmentStatus;
  senderId: APIRelatedResponse<APIResponse<UserModel>>;
  receiverId: APIRelatedResponse<APIResponse<UserModel>>;
}

export type AppointmentResponse = APIResponse<AppointmentModel>;

export type AppointmentDataResponse = {
  appointment: AppointmentResponse | null;
  error: string | null;
};

export type AppointmentsDataResponse = Promise<
  { appointments: AppointmentResponse[]; error: string | null } & MetaResponse
>;

export type AppointmentsResponse = {
  data: AppointmentResponse[];
  meta: MetaResponse;
};

export type AppointmentPayload = Partial<
  Omit<AppointmentModel, 'senderId' | 'receiverId'>
> & {
  senderId?: string;
  receiverId?: string;
};
