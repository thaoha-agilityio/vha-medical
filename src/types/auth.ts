import { UserSession } from './user';

export interface LoginFormData {
  identifier: string;
  password: string;
  remember: boolean;
}

export interface SignupFormData {
  username: string;
  email: string;
  password: string;
  confirmPassWord: string;
}

export type AuthResponse = {
  jwt: string;
  user: UserSession | null;
  error: string | null;
};

export interface RolePermission {
  id: number;
  name: string;
}

export type RolesResponse = {
  roles: RolePermission[];
  error: string | null;
};
