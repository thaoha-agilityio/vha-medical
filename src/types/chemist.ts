import {
  APIRelatedResponse,
  APIResponse,
  MetaResponse,
  UserModel,
} from '@/types';

export type ChemistModel = {
  users_permissions_user: APIRelatedResponse<APIResponse<UserModel>>;
};

export type ChemistResponse = APIResponse<ChemistModel>;

export type ChemistDataResponse = {
  chemist: ChemistResponse | null;
  error: string | null;
};

export type ChemistsDataResponse = Promise<
  { chemists: ChemistResponse[]; error: string | null } & MetaResponse
>;

export type ChemistsResponse = {
  data: ChemistResponse[];
  meta: MetaResponse;
};

export type ChemistPayload = {
  users_permissions_user: string;
};

export interface ChemistFormData {
  username: string;
  email: string;
  password: string;
  confirmPassWord: string;
  avatar?: string;
  description?: string;
  rating?: number;
  tasks?: number;
  reviews?: number;
  specialtyId: string;
}
