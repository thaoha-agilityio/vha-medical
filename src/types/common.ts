// Components
export type CustomClassType = {
  customClass?: string;
};

export type PageErrorProps = {
  error: Error & { digest?: string };
  reset: () => void;
};

// Status
export enum STATUS_TYPE {
  SUCCESS = 'success',
  WARNING = 'warning',
  ERROR = 'error',
}

// Theme mode
export enum THEME_MODE_TYPE {
  DARK = 'dark',
  LIGHT = 'light',
}

// Option
export interface Option {
  key: string;
  label: string;
}

// FetchData
export interface FetchDataProps {
  searchParams?: URLSearchParams;
  options?: RequestInit;
}
