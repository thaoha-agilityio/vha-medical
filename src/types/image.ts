export type ImageExtension = 'webp' | 'png';

export type ImageInfo = {
  filename: string;
  name: string;
  mime: string;
  extension: string;
  url: string;
};

export type ImageData = {
  id: string;
  title: string;
  url_viewer: string;
  url: string;
  display_url: string;
  width: string;
  height: string;
  size: string;
  time: string;
  expiration: string;
  image: ImageInfo;
  thumb: ImageInfo;
  medium: ImageInfo;
  delete_url: string;
};

export type ImgBBResponse = {
  data: ImageData;
  success: boolean;
  status: number;
};
