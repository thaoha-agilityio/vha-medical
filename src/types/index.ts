export * from './appointment';
export * from './auth';
export * from './chemist';
export * from './common';
export * from './image';
export * from './notification';
export * from './response';
export * from './searchParams';
export * from './specialty';
export * from './table';
export * from './user';
