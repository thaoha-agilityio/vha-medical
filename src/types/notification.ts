import {
  APIRelatedResponse,
  APIResponse,
  AppointmentStatus,
  MetaResponse,
  UserModel,
} from '@/types';

export interface InfoModel {
  id: string;
  status: AppointmentStatus;
  startTime: string;
  durationTime: string;
  content: string;
}

export interface NotificationModel {
  senderName: string;
  senderAvatar: string;
  isRead: boolean;
  info: InfoModel;
  senderId?: APIRelatedResponse<APIResponse<UserModel>>;
  createdAt?: string;
}

export type NotificationResponse = APIResponse<NotificationModel>;

export type NotificationDataResponse = {
  notification: NotificationResponse | null;
  error: string | null;
};

export type NotificationsDataResponse = Promise<
  { notifications: NotificationResponse[]; error: string | null } & MetaResponse
>;

export type NotificationsResponse = {
  data: NotificationResponse[];
  meta: MetaResponse;
};

export type NotificationPayload = Omit<NotificationModel, 'senderId'> & {
  senderId: string;
};
