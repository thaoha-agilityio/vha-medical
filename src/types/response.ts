export type APIResponse<T> = { id: string; attributes: T };

export type APIRelatedResponse<T> = { data: T };

export type MetaResponse = {
  pagination?: {
    page: number;
    pageSize: number;
    pageCount: number;
    total: number;
  };
};

export type TDataResponse<T> = {
  data: APIResponse<T>[];
  meta?: MetaResponse;
};

export interface ErrorResponse {
  data?: null;
  error: {
    status?: number;
    name?: string;
    message: string;
    details?: Record<string, unknown>;
  };
}
