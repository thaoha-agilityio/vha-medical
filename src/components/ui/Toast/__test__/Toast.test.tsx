import { render, screen, fireEvent } from '@testing-library/react';

// Types
import { STATUS_TYPE } from '@/types';

// Components
import { Toast } from '..';

jest.mock('@/utils', () => ({
  cn: jest.fn(() => 'mocked-class'),
}));

describe('Toast component', () => {
  const mockOnClose = jest.fn();

  it('renders the success toast correctly', () => {
    const { container } = render(
      <Toast
        message="Item moved successfully."
        status={STATUS_TYPE.SUCCESS}
        onClose={mockOnClose}
      />,
    );

    expect(screen.getByText('Item moved successfully.')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders the error toast correctly', () => {
    const { container } = render(
      <Toast
        message="Item has been deleted."
        status={STATUS_TYPE.ERROR}
        onClose={mockOnClose}
      />,
    );

    expect(screen.getByText('Item has been deleted.')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders the warning toast correctly', () => {
    const { container } = render(
      <Toast
        message="Improve password difficulty."
        status={STATUS_TYPE.WARNING}
        onClose={mockOnClose}
      />,
    );

    expect(
      screen.getByText('Improve password difficulty.'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('calls onClose when close button is clicked', () => {
    render(
      <Toast
        message="Item moved successfully."
        status={STATUS_TYPE.SUCCESS}
        onClose={mockOnClose}
      />,
    );

    const closeButton = screen.getByRole('button');
    fireEvent.click(closeButton);

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });
});
