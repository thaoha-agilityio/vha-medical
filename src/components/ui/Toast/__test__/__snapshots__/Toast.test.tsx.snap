// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component renders the error toast correctly 1`] = `
<div>
  <div
    class="overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex flex-row fixed z-[999] items-center justify-between w-full max-w-xs p-4 rounded-lg top-5 right-5 bg-background-100"
    data-testid="toast"
    role="alert"
    tabindex="-1"
  >
    <svg
      class="mocked-class"
      fill="none"
      viewBox="0 0 22 22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="11.2681"
        cy="10.7317"
        fill="white"
        r="7.5122"
      />
      <path
        d="M11 0C8.82441 0 6.69767 0.645139 4.88873 1.85383C3.07979 3.06253 1.66989 4.78049 0.83733 6.79048C0.00476611 8.80047 -0.213071 11.0122 0.211367 13.146C0.635804 15.2798 1.68345 17.2398 3.22183 18.7782C4.76021 20.3165 6.72022 21.3642 8.85401 21.7886C10.9878 22.2131 13.1995 21.9952 15.2095 21.1627C17.2195 20.3301 18.9375 18.9202 20.1462 17.1113C21.3549 15.3023 22 13.1756 22 11C21.9944 8.08433 20.8337 5.28969 18.772 3.228C16.7103 1.16631 13.9157 0.00559093 11 0ZM14.9875 13.7817C15.1464 13.9422 15.2355 14.1588 15.2355 14.3846C15.2355 14.6104 15.1464 14.8271 14.9875 14.9875C14.8258 15.1439 14.6096 15.2313 14.3846 15.2313C14.1596 15.2313 13.9435 15.1439 13.7817 14.9875L11 12.1952L8.21827 14.9875C8.05654 15.1439 7.84036 15.2313 7.61539 15.2313C7.39041 15.2313 7.17424 15.1439 7.0125 14.9875C6.85361 14.8271 6.76447 14.6104 6.76447 14.3846C6.76447 14.1588 6.85361 13.9422 7.0125 13.7817L9.80481 11L7.0125 8.21827C6.87757 8.05386 6.80862 7.84515 6.81905 7.63272C6.82948 7.42029 6.91856 7.21934 7.06895 7.06895C7.21934 6.91856 7.4203 6.82948 7.63272 6.81905C7.84515 6.80861 8.05387 6.87757 8.21827 7.0125L11 9.80481L13.7817 7.0125C13.9461 6.87757 14.1549 6.80861 14.3673 6.81905C14.5797 6.82948 14.7807 6.91856 14.9311 7.06895C15.0814 7.21934 15.1705 7.42029 15.181 7.63272C15.1914 7.84515 15.1224 8.05386 14.9875 8.21827L12.1952 11L14.9875 13.7817Z"
        fill="currentColor"
      />
    </svg>
    <p
      class="text-sm text-primary-200 max-w-[220px] ml-2 mr-3"
    >
      Item has been deleted.
    </p>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium !gap-0 transition-transform-colors-opacity motion-reduce:transition-none w-10 h-10 data-[hover=true]:opacity-hover border-transparent font-semibold text-primary-200 bg-none min-w-5 p-0"
      type="button"
    >
      <svg
        class="w-3 h-3"
        fill="none"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1l6 6m0 0l6 6M7 7l6-6M7 7l-6 6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Toast component renders the success toast correctly 1`] = `
<div>
  <div
    class="overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex flex-row fixed z-[999] items-center justify-between w-full max-w-xs p-4 rounded-lg top-5 right-5 bg-background-100"
    data-testid="toast"
    role="alert"
    tabindex="-1"
  >
    <svg
      aria-hidden="true"
      class="mocked-class"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
      />
    </svg>
    <p
      class="text-sm text-primary-200 max-w-[220px] ml-2 mr-3"
    >
      Item moved successfully.
    </p>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium !gap-0 transition-transform-colors-opacity motion-reduce:transition-none w-10 h-10 data-[hover=true]:opacity-hover border-transparent font-semibold text-primary-200 bg-none min-w-5 p-0"
      type="button"
    >
      <svg
        class="w-3 h-3"
        fill="none"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1l6 6m0 0l6 6M7 7l6-6M7 7l-6 6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Toast component renders the warning toast correctly 1`] = `
<div>
  <div
    class="overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex flex-row fixed z-[999] items-center justify-between w-full max-w-xs p-4 rounded-lg top-5 right-5 bg-background-100"
    data-testid="toast"
    role="alert"
    tabindex="-1"
  >
    <svg
      aria-hidden="true"
      class="mocked-class"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5ZM10 15a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-4a1 1 0 0 1-2 0V6a1 1 0 0 1 2 0v5Z"
      />
    </svg>
    <p
      class="text-sm text-primary-200 max-w-[220px] ml-2 mr-3"
    >
      Improve password difficulty.
    </p>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium !gap-0 transition-transform-colors-opacity motion-reduce:transition-none w-10 h-10 data-[hover=true]:opacity-hover border-transparent font-semibold text-primary-200 bg-none min-w-5 p-0"
      type="button"
    >
      <svg
        class="w-3 h-3"
        fill="none"
        viewBox="0 0 14 14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 1l6 6m0 0l6 6M7 7l6-6M7 7l-6 6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;
